plutus-tx-plugin/src/PlutusTx/Compiler/Kind.hs

```diff
@@ -16,7 +17,7 @@ import GHC.Plugins qualified as GHC
 import PlutusCore qualified as PLC
 
 compileKind :: Compiling uni fun m ann => GHC.Kind -> m (PLC.Kind ())
-compileKind k = withContextM 2 (sdToTxt $ "Compiling kind:" GHC.<+> GHC.ppr k) $ case k of
+compileKind k = withContextM 2 (sdToTxt msg) . traceCompilation msg $ case k of
```

Comment:
The tracking of the context here and the tracking of the parent steps seems somewhat redundant and could probably be merged?